src/util/serialize.cpp

```diff
@@ -120,121 +119,148 @@ std::string deSerializeString32(std::istream &is)
 }
 
 ////
-//// JSON
+//// JSON-like strings
 ////
 
 std::string serializeJsonString(const std::string &plain)
 {
-	std::ostringstream os(std::ios::binary);
-	os << "\"";
+	std::string tmp;
+
+	tmp.reserve(plain.size() + 2);
+	tmp.push_back('"');
 
 	for (char c : plain) {
 		switch (c) {
 			case '"':
-				os << "\\\"";
+				tmp.append("\\\"");
 				break;
 			case '\\':
-				os << "\\\\";
-				break;
-			case '/':
-				os << "\\/";
+				tmp.append("\\\\");
 				break;
 			case '\b':
-				os << "\\b";
+				tmp.append("\\b");
 				break;
 			case '\f':
-				os << "\\f";
+				tmp.append("\\f");
 				break;
 			case '\n':
-				os << "\\n";
+				tmp.append("\\n");
 				break;
 			case '\r':
-				os << "\\r";
+				tmp.append("\\r");
 				break;
 			case '\t':
-				os << "\\t";
+				tmp.append("\\t");
 				break;
 			default: {
 				if (c >= 32 && c <= 126) {
-					os << c;
+					tmp.push_back(c);
 				} else {
-					u32 cnum = (u8)c;
-					os << "\\u" << std::hex << std::setw(4)
-						<< std::setfill('0') << cnum;
+					// We pretend that Unicode codepoints map to bytes (they don't)
+					u8 cnum = static_cast<u8>(c);
+					tmp.append("\\u00");
+					tmp.push_back(hex_chars[cnum >> 4]);
+					tmp.push_back(hex_chars[cnum & 0xf]);
 				}
 				break;
 			}
 		}
 	}
 
-	os << "\"";
-	return os.str();
+	tmp.push_back('"');
+	return tmp;
+}
+
+static void deSerializeJsonString(std::string &s)
+{
+	assert(s.size() >= 2);
+	assert(s.front() == '"' && s.back() == '"');
+
+	size_t w = 0; // write index
+	size_t i = 1; // read index
+	const size_t len = s.size() - 1; // string length with trailing quote removed
+
+	while (i < len) {
+		char c = s[i++];
+		assert(c != '"');
```

Comment:
What about escaped quotation marks within the string?

Reply:
They're handled below and skipped over (this line does not run for every character in the input)